controller/controller.go

```diff
@@ -226,6 +229,39 @@ func ListenAndServeEx(host string, port int, dir string, ln *net.Listener, http
 	return server.ListenAndServe(host, port, protected, handler, opened, closed, ln, http)
 }
 
+func (c *Controller) watchGC() {
+	var elapsed uint64
```

Comment:
Using an `uint64` rather than a `time.Time` will work, but may cause a small amount of negative drift over time because the likelihood that each iteration will take slightly longer that one second. This will cause long `autogc` value such as 10 minutes to execute more often than every 10 minutes.

To work around this, I recommend using a timestamp and comparing current time against the previous time and triggering the GC operation when the duration has exceeded the duration.